### PR TITLE
Feat/nest views under dash

### LINKED ISSUE #2100: Clone Cell does not work
As a user with editor privileges or more,
when I click on clone cell on the corner of a cell in a dashboard
I create a new cell that is a clone of the existing cell
And if cloning fails, I get an error instead of a wrong success message
(right now the new cell is not created, but there is a success message anyways)

### PR DESCRIPTION
**Pull request by ebb-tide**
Closes #2100 

_Briefly describe your proposed changes:_
update FE to use new API for views that is nested under dashboards. 
Add cellID and dashboardID fields to views, 
Add dashboardID field to cells

### CODE REVIEW

File: http/cur_swagger.yml [hunk 1]
```diff
@@ -1407,11 +1491,11 @@ paths:
                   links:
                     $ref: "#/components/schemas/Links"
         default:
-          description: unexpected error
-          content:
-            application/json:
-              schema:
-                $ref: "#/components/schemas/Error"
+            description: unexpected error
```
Comment (OfTheDelmer) [inline http/cur_swagger.yml]: Is this tabbing off?

File: ui/src/dashboards/actions/v2/views.ts [hunk 2]
```diff
@@ -22,37 +22,37 @@ export interface SetViewAction {
...
+    dispatch(setView(cellID, null, RemoteDataState.Error))
   }
 }
 
-export const updateView = (view: View) => async (
+export const updateView = (dashboardID: string, view: View) => async (
   dispatch: Dispatch<Action>
-): Promise<void> => {
-  dispatch(setView(view.id, null, RemoteDataState.Loading))
+): Promise<View> => {
```
Comment (OfTheDelmer) [inline ui/src/dashboards/actions/v2/views.ts]: Nice

File: ui/src/dashboards/apis/v2/index.ts [hunk 3]
```diff
@@ -2,33 +2,54 @@
...
 // Types
-import {Dashboard, Cell, CreateCell} from 'src/api'
-import {DashboardSwitcherLinks} from 'src/types/v2/dashboards'
+import {
+  DashboardSwitcherLinks,
+  View,
+  Cell,
+  NewCell,
+  Dashboard,
+} from 'src/types/v2/'
```
Comment (OfTheDelmer) [inline ui/src/dashboards/apis/v2/index.ts]: trailing slash

File: ui/src/dashboards/apis/v2/index.ts [hunk 4]
```diff
@@ -2,33 +2,54 @@
...
+    return {...c, dashboardID}
+  })
+}
+
 // TODO(desa): what to do about getting dashboards from another v2 source
 export const getDashboards = async (): Promise<Dashboard[]> => {
   const {data} = await dashboardsAPI.dashboardsGet()
 
-  return data.dashboards
+  const dashboards = data.dashboards
```
Comment (OfTheDelmer) [inline ui/src/dashboards/apis/v2/index.ts]: is this `const` necessary

File: ui/src/dashboards/components/DashboardPage.tsx [hunk 5]
```diff
@@ -378,7 +382,7 @@ class DashboardPage extends Component<Props, State> {
   }
 }
 
-const mstp = (state, {params: {dashboardID}}): StateProps => {
+const mstp = (state: AppState, {params: {dashboardID}}): StateProps => {
```
Comment (OfTheDelmer) [inline ui/src/dashboards/components/DashboardPage.tsx]: nice!

File: ui/src/logs/actions/index.ts [hunk 6]
```diff
@@ -2,7 +2,7 @@
 import _ from 'lodash'
 import uuid from 'uuid'
 import {getDeep} from 'src/utils/wrappers'
-import {serverToUIConfig, uiToServerConfig} from 'src/logs/utils/config'
+import {serverToUIConfig} from 'src/logs/utils/config'
```
Comment (OfTheDelmer) [inline ui/src/logs/actions/index.ts]: ahh it's tough to have to update this when it's not part of alpha

File: ui/src/logs/actions/index.ts [hunk 7]
```diff
@@ -348,43 +343,54 @@ export const getLogConfigAsync = () => async (
...
   const state = getState()
   const isTruncated = getIsTruncated(state)
-  const views = await readViewsAJAX()
-  const logView: NewView | View = getDeep(views, '0', defaultLogView)
 
-  const logConfig = {
-    ...serverToUIConfig(logView),
-    isTruncated,
-  }
+  // TODO: fix this before adding logs back in to platform
```
Comment (OfTheDelmer) [inline ui/src/logs/actions/index.ts]: Sounds good!

**Comment by mark-rushakoff** [comment]
Done in https://github.com/influxdata/influxdb/pull/10959.

**Pull request closed without merging**